crates/vsock/src/vhu_vsock_thread.rs

```diff
@@ -462,76 +462,81 @@ impl VhostUserVsockThread {
-                    0
-                }
-            };
+                    Err(e) => {
+                        warn!("vsock: RX queue error: {:?}", e);
+                        0
+                    }
+                };
 
-            let vring = vring.clone();
-            let event_idx = self.event_idx;
+                let vring = vring.clone();
+                let event_idx = self.event_idx;
 
-            self.pool.spawn_ok(async move {
-                // TODO: Understand why doing the following in the pool works
-                if event_idx {
-                    if vring.add_used(head_idx, used_len as u32).is_err() {
-                        warn!("Could not return used descriptors to ring");
-                    }
-                    match vring.needs_notification() {
-                        Err(_) => {
-                            warn!("Could not check if queue needs to be notified");
-                            vring.signal_used_queue().unwrap();
+                self.pool.spawn_ok(async move {
+                    // TODO: Understand why doing the following in the pool works
+                    if event_idx {
+                        if vring.add_used(head_idx, used_len as u32).is_err() {
+                            warn!("Could not return used descriptors to ring");
                         }
-                        Ok(needs_notification) => {
-                            if needs_notification {
+                        match vring.needs_notification() {
+                            Err(_) => {
+                                warn!("Could not check if queue needs to be notified");
                                 vring.signal_used_queue().unwrap();
                             }
+                            Ok(needs_notification) => {
+                                if needs_notification {
+                                    vring.signal_used_queue().unwrap();
+                                }
+                            }
                         }
+                    } else {
+                        if vring.add_used(head_idx, used_len as u32).is_err() {
+                            warn!("Could not return used descriptors to ring");
+                        }
+                        vring.signal_used_queue().unwrap();
                     }
-                } else {
-                    if vring.add_used(head_idx, used_len as u32).is_err() {
-                        warn!("Could not return used descriptors to ring");
-                    }
-                    vring.signal_used_queue().unwrap();
-                }
-            });
+                });
 
-            match rx_queue_type {
-                RxQueueType::Standard => {
-                    if !self.thread_backend.pending_rx() {
-                        break;
+                match rx_queue_type {
+                    RxQueueType::Standard => {
+                        if !self.thread_backend.pending_rx() {
+                            break;
```

Comment:
Same here, I think the original idea was to stop processing new descriptors, since we don't have any packet to copy in it.